src/test/java/org/takes/misc/VerboseListTest.java

```diff
@@ -0,0 +1,407 @@
+        this.thrown.expectMessage(MSG);
+        this.thrown.expectCause(Matchers.is(cause));
+        this.list.get(index);
+    }
+
+    /**
+     * VerboseList should wraps OutOfBoundsException thrown by set method.
+     * @throws Exception If something goes wrong.
+     */
+    @Test
+    public final void wrapsIndexOutOfBoundsExceptionFromSet() throws Exception {
+        final int index = 5;
+        final Object obj = new Object();
+        final Exception cause = new IndexOutOfBoundsException();
+        Mockito.doThrow(cause).when(this.origin).set(index, obj);
+        this.thrown.expect(IllegalArgumentException.class);
+        this.thrown.expectMessage(MSG);
+        this.thrown.expectCause(Matchers.is(cause));
+        this.list.set(index, obj);
+    }
+
+    /**
+     * VerboseList should wraps OutOfBoundsException thrown by add method.
+     * @throws Exception If something goes wrong.
+     */
+    @Test
+    public final void wrapsIndexOutOfBoundsExceptionFromAdd() throws Exception {
+        final int index = 5;
+        final Object obj = new Object();
+        final Exception cause = new IndexOutOfBoundsException();
+        Mockito.doThrow(cause).when(this.origin).add(index, obj);
+        this.thrown.expect(IllegalArgumentException.class);
+        this.thrown.expectMessage(MSG);
+        this.thrown.expectCause(Matchers.is(cause));
+        this.list.add(index, obj);
+    }
+
+    /**
+     * VerboseList should wraps OutOfBoundsException thrown by remove method.
+     * @throws Exception If something goes wrong.
+     */
+    @Test
+    public final void wrapsIndexOutOfBoundsExceptionFromRemove()
+        throws Exception {
+        final int index = 5;
+        final Exception cause = new IndexOutOfBoundsException();
+        Mockito.doThrow(cause).when(this.origin).remove(index);
+        this.thrown.expect(IllegalArgumentException.class);
+        this.thrown.expectMessage(MSG);
+        this.thrown.expectCause(Matchers.is(cause));
+        this.list.remove(index);
+    }
+
+    /**
+     * VerboseList should wraps OutOfBoundsException thrown by listIterator
+     * method.
+     * @throws Exception If something goes wrong.
+     */
+    @Test
+    public final void wrapsIndexOutOfBoundsExceptionFromListIterator()
+        throws Exception {
+        final int index = 5;
+        final Exception cause = new IndexOutOfBoundsException();
+        Mockito.doThrow(cause).when(this.origin).listIterator(index);
+        this.thrown.expect(IllegalArgumentException.class);
+        this.thrown.expectMessage(MSG);
+        this.thrown.expectCause(Matchers.is(cause));
+        this.list.listIterator(index);
+    }
+
+    /**
+     * VerboseList should wraps OutOfBoundsException thrown by subList method.
+     * @throws Exception If something goes wrong.
+     */
+    @Test
+    public final void wrapsIndexOutOfBoundsExceptionFromSubList()
+        throws Exception {
+        final int frmidx = 2;
```

Comment:
@Happy-Neko Let's call this `from`, [avoid compound names](http://www.yegor256.com/2015/01/12/compound-name-is-code-smell.html).